<?xml version="1.0"?>
<launch>
  <arg name="use_rviz" default="true" />
  <arg name="use_moveit" default="true" />
  <arg name="robot_ip" default="192.168.0.4" />

  <!-- joint group pos control -->
  <!-- <arg name="controllers" default="joint_state_controller joint_group_pos_controller" />
  <arg name="stopped_controllers" default="cartesian_motion_controller motion_control_handle" /> -->

  <!-- cartesian motion control -->
  <!-- <arg name="controllers" default="joint_state_controller cartesian_motion_controller motion_control_handle" />
  <arg name="stopped_controllers" default="joint_group_pos_controller" /> -->

  <!-- joint traj control -->
  <arg name="controllers" default="joint_state_controller pos_joint_traj_controller" />
  <arg name="stopped_controllers" default="joint_group_pos_controller cartesian_motion_controller motion_control_handle" />

  <!-- Set the robot urdf on the parameter server -->
  <arg name="urdf_path" default="$(find aubo_i5_description)/urdf/aubo_i5.xacro" />
  <param name="robot_description" command="xacro '$(arg urdf_path)'" />

  <!-- Load controller settings -->
  <arg name="controller_config_file" default="$(find aubo_ros_control)/config/aubo_controllers.yaml" />
  <rosparam file="$(arg controller_config_file)" command="load" />

  <node name="aubo_hardware_interface" pkg="aubo_ros_control" type="aubo_hardware_interface_node"
    output="screen">
    <param name="robot_ip" type="str" value="$(arg robot_ip)" />
  </node>

  <!-- spawn controller manager -->
  <node name="ros_control_controller_spawner" pkg="controller_manager" type="spawner"
    respawn="false"
    output="screen" args="$(arg controllers)" />

  <!-- load other controller -->
  <node name="ros_control_stopped_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" args="--stopped $(arg stopped_controllers)"
    unless="$(eval arg('stopped_controllers') == '')" />

  <!-- Robot state publisher -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="50.0" />
    <param name="tf_prefix" type="string" value="" />
  </node>

  <!-- Load moveit -->
  <group if="$(arg use_moveit)">
    <include file="$(find aubo_i5_moveit_config)/launch/move_group.launch">
      <arg name="load_robot_description" value="false"/>
    </include>
  </group>

  <group if="$(arg use_rviz)">
    <arg name="my_rviz" value="-d $(find aubo_ros_control)/config/rviz.rviz" />
    <node name="rviz" pkg="rviz" type="rviz" respawn="false" args="$(arg my_rviz)" output="screen" />
  </group>


</launch>